modules/serverpod_auth/serverpod_auth_server/lib/src/business/email_auth.dart

```diff
@@ -13,13 +13,13 @@ class Emails {
   /// Generates a password hash from a users password and email. This value
   /// can safely be stored in the database without the risk of exposing
   /// passwords.
-  static Future<String> generatePasswordHash(String password) async {
-    return PasswordHash.argon2id(
-      password,
-      pepper: EmailSecrets.pepper,
-      allowUnsecureRandom: AuthConfig.current.allowUnsecureRandom,
-    );
-  }
+  static Future<String> generatePasswordHash(String password) async =>
+      AuthConfig.current.generatePasswordHashCallback?.call(password) ??
+      PasswordHash.argon2id(
+        password,
+        pepper: EmailSecrets.pepper,
+        allowUnsecureRandom: AuthConfig.current.allowUnsecureRandom,
+      );
```

Comment:
Suggestion: assign this directly in the auth configuration.

This removes a redundant null check here.